public/app/features/explore/TraceView/createSpanLink.tsx

```diff
@@ -80,35 +82,66 @@ function legacyCreateSpanLinkFactory(splitOpenFn: SplitOpen, traceToLogsOptions?
     // the moment. Issue is that the trace itself isn't clearly mapped to dataFrame (right now it's just a json blob
     // inside a single field) so the dataLinks as config of that dataFrame abstraction breaks down a bit and we do
     // it manually here instead of leaving it for the data source to supply the config.
+    const query = getLinkForSplunk(span, traceToLogsOptions);
+    const expr = getLinkForLoki(span, traceToLogsOptions);
 
-    const expr = getLokiQueryFromSpan(span, traceToLogsOptions);
-    if (!expr) {
+    if (!expr && dataSourceSettings?.type === 'loki') {
       return undefined;
     }
 
-    const dataLink: DataLink<LokiQuery> = {
-      title: dataSourceSettings.name,
-      url: '',
-      internal: {
-        datasourceUid: dataSourceSettings.uid,
-        datasourceName: dataSourceSettings.name,
-        query: {
-          expr,
-          refId: '',
-        },
-      },
-    };
+    let dataLink: DataLink<LokiQuery | DataQuery> = {} as DataLink<LokiQuery | DataQuery>;
+
+    switch (dataSourceSettings?.type) {
+      case 'loki':
+        getLinkForLoki(span, traceToLogsOptions);
+        dataLink = {
```

Comment:
Could consider moving building the datalink into the `getLinkForX` functions so they return the datalink instead of just the query so we'd then have `dataLink = getLinkForLoki(span, traceToLogsOptions)`